mlflow/tensorflow/__init__.py

```diff
@@ -407,7 +407,7 @@ def save_model(
         # To maintain prior behavior, when the format is HDF5, we save
         # with the h5 file extension. Otherwise, model_path is a directory
         # where the saved_model.pb will be stored (for SavedModel format)
-        file_extension = ".h5" if save_format == "h5" else ""
+        file_extension = ".h5" if save_format == "h5" else ".keras"
```

Comment:
We might want to consider dropping support for `.h5` if it's legacy.

Reply:
To make those tests pass for all versions, it seems we need to refactor, no simple fix. I'll close this PR as we plan to address them altogether soon.

Reply:
sg